src/com/stanimiriliev/chaptertwo/homogeneoustree/Tree.java

```diff
@@ -18,97 +18,141 @@ public Tree(int root) {
 
```

Comment:
the "correct" is very interesting as description. Your first solution works as expected, I just want from you to redesign some of the part, because like that you look over the task from other points. More code(excesses) = more experience.


I think now the code looks better and cleaner.